projectile.el

```diff
@@ -4137,6 +4140,32 @@ Otherwise behave as if called interactively.
 ;;;###autoload
 (define-obsolete-function-alias 'projectile-global-mode 'projectile-mode "1.0")
 
+(defun projectile--repeat-until-project-buffer (orig-fun &rest args)
+  "Repeat ORIG-FUN with ARGS until the current buffer is a project buffer."
+  (let* ((other-project-buffers (make-hash-table :test 'eq))
+         (projectile-project-buffers (projectile-project-buffers))
+         (max-iterations (length (buffer-list)))
+         (counter 0))
+    (dolist (buffer projectile-project-buffers)
+      (unless (eq buffer (current-buffer))
+        (puthash buffer t other-project-buffers)))
+    (when (cdr-safe projectile-project-buffers)
+      (while (and (< counter max-iterations)
+                  (not (gethash (current-buffer) other-project-buffers)))
+        (apply orig-fun args)
+        (incf counter)))))
+
+(defun projectile-next-project-buffer ()
+  "In selected window switch to the next project buffer."
+  (interactive)
+  (projectile--repeat-until-project-buffer #'next-buffer))
```

Comment:
How does the command behave when invoked outside a project? I guess it should just show some error as suggesting to switch to a project in this case doesn't make much sense.

Reply:
alternatively, it may (warn and) forward to previous-buffer / next-buffer

Reply:
Yeah, that might actually be even better.